From e26c1ca12e1b3b471063e37d5e8e287549a6da22 Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:32:12 -0400
Subject: [x86] avx-512: Enable AVX-512 States Context Switch

Message-id: <1408595532-28320-3-git-send-email-ruwang@redhat.com>
Patchwork-id: 88503
O-Subject: [PATCH RHEL7.1 BZ1085563 2/2] x86, AVX-512: Enable AVX-512 States Context Switch
Bugzilla: 1085563
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

From: "Wang, Xiaolong" <xiaolong.wang@intel.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1085563

Backport of:
Original commit c2bc11f10a39527cd1bb252097b5525664560956
Original Author: Fenghua Yu <fenghua.yu@intel.com>
Original Date:   Thu Feb 20 13:24:51 2014 -0800

    x86, AVX-512: Enable AVX-512 States Context Switch

    This patch enables Opmask, ZMM_Hi256, and Hi16_ZMM AVX-512 states for
    xstate context switch.

Signed-off-by: Wang, Xiaolong <xiaolong.wang@intel.com>
Signed-off-by: Wang, Rui <ruwang@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/xsave.h b/arch/x86/include/asm/xsave.h
index dcd047b..923088f 100644
--- a/arch/x86/include/asm/xsave.h
+++ b/arch/x86/include/asm/xsave.h
@@ -9,6 +9,9 @@
 #define XSTATE_FP 0x1
 #define XSTATE_SSE 0x2
 #define XSTATE_YMM 0x4
+#define XSTATE_OPMASK  0x20
+#define XSTATE_ZMM_Hi256 0x40
+#define XSTATE_Hi16_ZMM  0x80
 #define XSTATE_BNDREGS 0x8
 #define XSTATE_BNDCSR 0x10
 
@@ -31,7 +34,8 @@
 #define XSTATE_EAGER (XSTATE_BNDREGS | XSTATE_BNDCSR)
 
 /* All currently supported features */
-#define XCNTXT_MASK (XSTATE_LAZY | XSTATE_EAGER)
+#define XCNTXT_MASK (XSTATE_FP | XSTATE_SSE | XSTATE_YMM  \
+   | XSTATE_OPMASK | XSTATE_ZMM_Hi256 | XSTATE_Hi16_ZMM)
 
 #ifdef CONFIG_X86_64
 #define REX_PREFIX "0x48, "
-- 
1.7.1